### PR TITLE
Fix issue that made redis release immutable

### PR DESCRIPTION
**Pull request by glrf**
We used the observed redis release to build the desired release. This completely dropped any changes from the composition. This PR fixes this by the desired released and only using the observed release to get the current version.

I also included a refactor as the initial bug likely got through review as currently the function is bit convoluted. 

## Checklist

- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [x] Update tests.

### CODE REVIEW

File: pkg/comp-functions/functions/vshnredis/release.go [hunk 1]
```diff
@@ -178,22 +194,18 @@ func getReleaseValues(r *v1beta1.Release) (map[string]interface{}, error) {
...
+func setReleaseVersion(comp *vshnv1.VSHNRedis, values map[string]interface{}, observed map[string]interface{}) error {
+	var err error
+	tag, _, err := unstructured.NestedString(observed, "image", "tag")
 	if err != nil {
 		return fmt.Errorf("cannot get image tag from values in release: %v", err)
 	}
-	// In case the tag is set then let the maintenance manage it
 	if tag != "" {
-		return nil
+		// In case the tag is set, keep the desired
```
Comment (Kidswiss) [inline pkg/comp-functions/functions/vshnredis/release.go:204]: You mean keep the observed?

Comment (glrf) [reply pkg/comp-functions/functions/vshnredis/release.go:204]: Yes. You're right I got confused updating these comments :sweat_smile:

**Review by Kidswiss — CHANGES_REQUESTED** [review]
Just one confusing comment I found, the rest LGTM

File: pkg/comp-functions/functions/vshnredis/release.go (hunk 1 above)
Comment (Kidswiss) [reply pkg/comp-functions/functions/vshnredis/release.go]: and I got confused reading the comment :D

**Review by Kidswiss — APPROVED** [review]

**Pull request merged**